### ISSUE TITLE
@SaveInstance works strange on fragments

### ISSUE THREAD

**Issue by eugene-kamenev** [Arasthel/SwissKnife#10]
tried to use @SaveInstance annotation inside fragment, when activity goes to pause, allocated space grows very large, and I see messages like:

Background partial concurrent mark sweep GC freed 26(1372KB) AllocSpace objects, 7(1358KB) LOS objects, 26% free, 11MB/15MB, paused 37.927ms total 157.183ms
01-17 00:36:30.269    2250-2262/com.coupledays.app I/art﹕ Background sticky concurrent mark sweep GC freed 20(1960KB) AllocSpace objects, 10(1939KB) LOS objects, 14% free, 11MB/13MB, paused 13.785ms total 66.584ms

@dexafree can you help with it?

**Comment by Arasthel** [comment]
I just found where the problem is, thought I don't really know how to fix this.

For some reason, when you generate via AST the "super.onSaveInstanceState(savedState)" statement, it doesn't really call "super." but "this.", which ends up in a StackOverflowException as shown here:

```
    --------- beginning of crash
01-16 22:40:14.967    2954-2954/com.coupledays.app E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.coupledays.app, PID: 2954
    java.lang.StackOverflowError: stack size 8MB
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1017)
            at org.codehaus.groovy.runtime.InvokerHelper.invokePojoMethod(InvokerHelper.java:902)
            at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:893)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodN(ScriptBytecodeAdapter.java:166)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at j
01-16 22:40:15.037    2954-2954/com.coupledays.app D/Error﹕ ERR: exClass=java.lang.StackOverflowError
01-16 22:40:15.037    2954-2954/com.coupledays.app D/Error﹕ ERR: exMsg=stack size 8MB
01-16 22:40:15.037    2954-2954/com.coupledays.app D/Error﹕ ERR: file=Method.java
01-16 22:40:15.037    2954-2954/com.coupledays.app D/Error﹕ ERR: class=java.lang.reflect.Method
01-16 22:40:15.037    2954-2954/com.coupledays.app D/Error﹕ ERR: method=invoke line=-2
01-16 22:40:15.054    2954-2969/com.coupledays.app W/art﹕ Suspending all threads took: 12.750ms
01-16 22:40:15.054    2954-2954/com.coupledays.app D/Error﹕ ERR: stack=java.lang.StackOverflowError: stack size 8MB
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1017)
            at org.codehaus.groovy.runtime.InvokerHelper.invokePojoMethod(InvokerHelper.java:902)
            at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:893)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodN(ScriptBytecodeAdapter.java:166)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
            at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
            at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1208)
            at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
            at com.coupledays.view.fragments.apartment.DefaultListFragment.onSaveInstanceState(DefaultListFragment.groovy)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang
01-16 22:40:15.055    2954-2954/com.coupledays.app D/Error﹕ ERR: TOTAL BYTES WRITTEN: 7605296
01-16 22:40:15.056    2954-2954/com.coupledays.app E/JavaBinder﹕ !!! FAILED BINDER TRANSACTION !!!
01-16 22:40:15.056    2954-2954/com.coupledays.app E/AndroidRuntime﹕ Error reporting crash
    android.os.TransactionTooLargeException
            at android.os.BinderProxy.transactNative(Native Method)
            at android.os.BinderProxy.transact(Binder.java:496)
            at android.app.ActivityManagerProxy.handleApplicationCrash(ActivityManagerNative.java:4100)
            at com.android.internal.os.RuntimeInit$UncaughtHandler.uncaughtException(RuntimeInit.java:89)
            at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:693)
            at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:690)
```

The problem is I have no idea why it is working that way. Maybe a groovy bug? Sounds like it would have been discovered already.

**Comment by eugene-kamenev** [comment]
fixed in 1.2.2
